### PR TITLE
ANDROID_HOME set from in UI

### PR DESCRIPTION
**Pull request by dpgraham**
* Added a 'Config' window that is either opened from menu or from the start server page
* Added form that sets environment variables JAVA_HOME and ANDROID_HOME
* Shows defaults as whatever 'appium-desktop' finds inside of `process.env` originally
* When these are saved, saves it to appium-settings
* When app is loaded for first time, `process.env` is extended with these saved env vars
* When config is 'Saved' the user is asked to 'Restart Now' for changes to take effect

* Other

  * Added 'Stubs' for actions, reducers and components for cutting-and-pasting purposes later on
  * Moved browser window opener to a helper method

* This is how it looks, we'll probably want to change it to something more subtle
screen shot 2018-07-24 at 4 12 52 pm

screen shot 2018-07-24 at 4 19 54 pm

### CODE REVIEW

**Comment by dpgraham** [comment]
@jlipps Did you have a chance to look at this?

Also, are there ENV variables other than ANDROID_HOME, or JAVA_HOME that should be configurable?

**Comment by jlipps** [comment]
@dpgraham trying to take a look now but i'm getting conflicts pulling the branch. can you fix then i'll play with it?

**Comment by dpgraham** [comment]
FIxed

**Comment by jlipps** [comment]
pulled and reviewing today

**Comment by jlipps** [comment]
when i pulled this and ran it for the first time, i got `ANDROID_HOME` not defined error on opening the config window

**Comment by jlipps** [comment]
seems like the config window should default to a much smaller size

File: app/main/appium.js [hunk 1]
```diff
@@ -374,6 +355,39 @@ function connectMoveToApplicationsFolder () {
...
+  });
+}
+
+function connectSaveEnv () {
+  ipcMain.on('appium-save-env', async (event, environmentVariables) => {
+    // Pluck unset values
+    const env = _.pickBy(environmentVariables, _.identity);
+
+    await settings.set('ENV', env);
+    setEnv();
```
Comment (jlipps) [inline app/main/appium.js]: `await`

File: app/renderer/actions/Stub.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const NAME_OF_ACTION_1 = 'NAME_OF_ACTION_1';
```
Comment (jlipps) [inline app/renderer/actions/Stub.js]: is this file needed?

File: app/renderer/components/Config/Config.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const {environmentVariables} = this.props;
+    ipcRenderer.send('appium-save-env', environmentVariables);
+    ipcRenderer.once('appium-save-env-done', () => {
+      const message = `Application must be restarted for changes to take effect`;
+      const dialogOptions = {type: 'info', buttons: ['Restart Now', 'Restart Later'], message};
+      dialog.showMessageBox(dialogOptions, (response) => {
+        if (response === 0) {
+          // If 'Restart Now' restart the application
+          app.relaunch();
+          app.exit();
```
Comment (jlipps) [inline app/renderer/components/Config/Config.js]: why do we both relaunch and exit?

File: app/renderer/components/Config/Config.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          getCurrentWindow().close();
+        }
+      });
+    });
+  }
+
+  render () {
+    const {setEnvironmentVariable, environmentVariables, defaultEnvironmentVariables} = this.props;
+
+    const ENV_VARIABLE_NAMES = [
```
Comment (jlipps) [inline app/renderer/components/Config/Config.js]: put these up top?

File: app/renderer/components/Stub.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React, { Component } from 'react';
```
Comment (jlipps) [inline app/renderer/components/Stub.js]: this stub file looks unnecessary

File: app/renderer/reducers/Stub.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import {
```
Comment (jlipps) [inline app/renderer/reducers/Stub.js]: unused file

File: app/renderer/actions/Stub.js (hunk 2 above)
Comment (dpgraham) [reply app/renderer/actions/Stub.js]: These are 'Stub' files for future usage for copying and pasting to create new actions/reducers/components

File: app/renderer/components/Config/Config.js (hunk 3 above)
Comment (dpgraham) [reply app/renderer/components/Config/Config.js]: Not sure why we have to do that but apparently it's required: https://github.com/electron/electron/issues/6579

File: app/renderer/actions/Stub.js (hunk 2 above)
Comment (jlipps) [reply app/renderer/actions/Stub.js]: oh i see so independently of this PR you want to have stub files for developer assistance in the future.

File: app/renderer/components/Config/Config.js (hunk 3 above)
Comment (jlipps) [reply app/renderer/components/Config/Config.js]: ok, as long as it's the way we're supposed to do it!

**Comment by jlipps** [comment]
hey @dpgraham i'm heading out on vacation, so from my perspective if you fix the issues mentioned here it's good to go

**Comment by mrk-han** [comment]
Does this mean that Appium Desktop is no longer reading from zshconfig/bash_profiles?

**Comment by dpgraham** [comment]
@mrk-han It works the same as before, the only difference is that if you opt to set an environment variable from the UI, it will use that value instead of what you have set in your bash profile.

**Comment by swarnock89** [comment]
@dpgraham This may not be working as intended. When I don't have paths set in the configuration menu, my tests don't start up. When I copy the paths from my .bash_profile and enter them in the configuration menu, my tests run fine.
